src/main/resources/db/migration/V0002__purge.sql

```diff
@@ -0,0 +1,259 @@
+    	VALUES (nextval('purge_job_audit_messages_seq'), i_purge_job_id, now(),
+				'Deleted individual_request : ' || c3_rec.sdt_request_reference);
+
+	    n_iteration := n_iteration + 1;
+    END LOOP;
+
+    RAISE NOTICE '% records deleted from individual_requests table.', n_iteration;
+
+    INSERT INTO sdt_owner.purge_job_audit_messages
+    (purge_job_message_id, purge_job_id, message_date,log_message)
+    VALUES (nextval('purge_job_audit_messages_seq'), i_purge_job_id, now(),
+	        n_iteration || ' records deleted from individual_requests table.');
+
+	-- update count of deletions - error logs
+    UPDATE sdt_owner.purge_job_audit
+  	SET count_of_individual_requests_deleted =n_iteration
+  	WHERE purge_job_id=i_purge_job_id;
+
+    RAISE NOTICE 'Commencing deletion of bulk_submissions records';
+
+    INSERT INTO sdt_owner.purge_job_audit_messages
+    (purge_job_message_id, purge_job_id, message_date,log_message)
+    VALUES (nextval('purge_job_audit_messages_seq'), i_purge_job_id, now(),
+			'Commencing deletion of bulk_submissions records');
+
+    n_iteration := 0;
+    FOR c2_rec IN c2 LOOP
+        DELETE FROM sdt_owner.bulk_submissions
+        WHERE bulk_submission_id = c2_rec.bulk_submission_id;
+
+        RAISE NOTICE 'Deleted bulk_submissions : %', c2_rec.bulk_submission_id;
+
+	    INSERT INTO sdt_owner.purge_job_audit_messages
+		(purge_job_message_id, purge_job_id, message_date,log_message)
+    	VALUES (nextval('purge_job_audit_messages_seq'), i_purge_job_id, now(),
+				'Deleted bulk_submissions : ' || c2_rec.bulk_submission_id);
+
+        n_iteration := n_iteration + 1;
+    END LOOP;
+
+    RAISE NOTICE '% records deleted from bulk_submissions table.', n_iteration;
+
+    INSERT INTO sdt_owner.purge_job_audit_messages
+    (purge_job_message_id, purge_job_id, message_date,log_message)
+    VALUES (nextval('purge_job_audit_messages_seq'), i_purge_job_id, now(),
+			n_iteration || ' records deleted from bulk_submissions table.');
+
+	UPDATE sdt_owner.purge_job_audit
+  	SET count_of_bulk_submissions_deleted=n_iteration
+  	WHERE purge_job_id=i_purge_job_id;
+
+    RAISE NOTICE 'Commencing deletion of service_requests records';
+
+    INSERT INTO sdt_owner.purge_job_audit_messages
+    (purge_job_message_id, purge_job_id, message_date,log_message)
+    VALUES (nextval('purge_job_audit_messages_seq'), i_purge_job_id, now(),
+	        'Commencing deletion of service_requests records');
+
+    n_iteration := 0;
+    FOR c1_rec IN c1 LOOP
+        DELETE FROM sdt_owner.service_requests
+        WHERE service_request_id = c1_rec.service_request_id;
+
+	    RAISE NOTICE 'Deleted service_requests : %', c1_rec.service_request_id;
+
+        INSERT INTO sdt_owner.purge_job_audit_messages
+    	(purge_job_message_id, purge_job_id, message_date,log_message)
+    	VALUES (nextval('purge_job_audit_messages_seq'), i_purge_job_id, now(),
+				'Deleted service_requests : ' || c1_rec.service_request_id);
```

Comment:
The original version of the procedure didn't log the deletion of individual service requests.  Any reason it's been added in this version?

Reply:
Gone